### PR TITLE
retrieve the commitStatus for a specific context

### PR DESCRIPTION
**Pull request by tomverin**

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported no new issues.

File: src/main/java/org/sonar/plugins/github/PullRequestFacade.java [hunk 1]
```diff
@@ -253,4 +254,21 @@ public String getGithubUrl(@Nullable InputPath inputPath, @Nullable Integer issu
...
   }
+
+  public GHCommitStatus getCommitStatusForContext(GHPullRequest pr, String context){
+    List<GHCommitStatus> statuses = null;
+    try {
+      statuses = pr.getRepository().listCommitStatuses(pr.getHead().getSha()).asList();
+    } catch (IOException e) {
+      throw new IllegalStateException("Unable to retrieve commit statuses.", e);
+    }
+    if  (!statuses.isEmpty()) {
```
Comment (henryju) [inline src/main/java/org/sonar/plugins/github/PullRequestFacade.java]: Not necessary when doing a foreach

File: src/main/java/org/sonar/plugins/github/PullRequestFacade.java [hunk 2]
```diff
@@ -253,4 +254,21 @@ public String getGithubUrl(@Nullable InputPath inputPath, @Nullable Integer issu
     }
     return null;
   }
+
+  public GHCommitStatus getCommitStatusForContext(GHPullRequest pr, String context){
```
Comment (henryju) [inline src/main/java/org/sonar/plugins/github/PullRequestFacade.java]: Why public? Need @CheckForNull

**Comment by SonarTech** [comment]
SonarQube analysis reported no new issues.

**Comment by SonarTech** [comment]
SonarQube analysis reported no new issues.

**Comment by henryju** [comment]
@tomverin could you please update your pull request to merge all your commits in a single "clean" one

**Comment by tomverin** [comment]
@henryju done

**Comment by SonarTech** [comment]
SonarQube analysis reported no new issues.

**Pull request merged**